app/assets/javascripts/google-analytics-events.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * Tracks custom events with Google Analytics.
+ * This application uses the google-analytics-rails gem to track general usage
+ * data. This file defines additional user interations to be tracked with Google
+ * Analtyics.
+ * See https://developers.google.com/analytics/devguides/collection/analyticsjs/events
+ */
+
+$(function(){
+
+  // Track source view when sources#show page loads.
+  $(document).ready(function(){
+    if ($('.sources-controller.show-action').length > 0) {
+      trackSourceViewEvent();
+    }
+  });
+
+  /*
+   * Bind event to a.ViewObject links to track click-throughs to external
+   * provider pages.  Assumes that all relevlant links have 'ViewObject' class.
+   */
+  $('body').on('click', '.ViewObject', function(){
+    event.preventDefault();
+    trackClickThroughEvent(this);
+    window.open(this.href, '_blank');
+  });
+});
+
+/* Track pageview of an internal source page.
+ * Relies on data-* attributes embedded in the HTML of the page.
+ */
+var trackSourceViewEvent = function(){
+  /* 
+   * 'ga' must be defined for a signal to be sent to Google Analtyics.
+   * It is set by the google analtyics rails gem if a valid tracker is present.
+   */
+  if (typeof ga !== 'undefined') {
+
+    var item = $("[data-item-id]");
+    var category = "View Primary Source : " + item.attr("data-provider");
+    var action = item.attr("data-data-provider");
+    var label = item.attr("data-item-id") + " : " + item.attr("data-title");
+
+    ga('send', 'event', category, action, label);
+  }
+}
+
+/* 
+ * Track click-throughs to external provider pages.
+ * @param HTML object to which the event is bound.
+ * Relies on data-* attributes belonging to a parent of the given object.
+ */
+var trackClickThroughEvent = function(obj){
+  /* 
+   * 'ga' must be defined for a signal to be sent to Google Analtyics.
+   * It is set by the google analtyics rails gem if a valid tracker is present.
+   */
+  if (typeof ga !== 'undefined') {
+
+    var item = $(obj).parents("[data-item-id]");
+    var category = "Click Through : " + item.attr("data-provider");
+    var action = item.attr("data-data-provider");
+    var label = item.attr("data-item-id") + " : " + item.attr("data-title");
+
+    ga('send', 'event', category, action, label, {'transport': 'beacon'});
```

Comment:
@markbreedlove I had originally used the `beacon` transport b/c pages were being unloaded when users clicked through to provider pages (ie. the links were opening in the same tab).  The `beacon` helps ensure that the data gets sent to Google Analytics on page unload.  However, part of this PR changes the links so that they [open in new tabs](https://github.com/dpla/primary-source-sets/pull/174/files#diff-b6a20150fda7fa0a5aef93e64f6a5c28R74).  This makes it un-necessary to use `beacon`, so I will remove it and just use the default transport method instead.